src/rest_vol.c

```diff
@@ -629,11 +629,16 @@ H5_rest_term(void)
     } /* end if */
 
     /* Cleanup type info array */
-    for (size_t i = 0; i < H5I_MAX_NUM_TYPES; i++) {
-        rv_hash_table_free(RV_type_info_array_g[i]->table);
-        RV_free(RV_type_info_array_g[i]);
-        RV_type_info_array_g[i] = NULL;
+    if (RV_type_info_array_g) {
```

Comment:
It's likely the fact that it _isn't_ explicitly initialized to NULL entries, so it ends up trying to free random pointers. The condition here probably doesn't do much since `RV_type_info_array_g` is an array of pointers. Rather, I think the real fix should be changing `RV_type_info *RV_type_info_array_g[H5I_MAX_NUM_TYPES];` to `RV_type_info *RV_type_info_array_g[H5I_MAX_NUM_TYPES] = {0};` at the top of this file.

Reply:
Note that freeing a NULL pointer is allowed by the standard, so this is almost certainly due to the array having uninitialized entries that the code is attempting to free.

Reply:
This particular issue was due to the top level `RV_type_info_array_g` pointer being NULL when referenced in line 633, so I think we should keep that check around. It's still true that we should avoid passing around uninitialized pointers - I've added the initialization.